pkg/sessionctx/stmtctx/stmtctx.go

```diff
@@ -1299,6 +1303,44 @@ func (sc *StatementContext) GetResultRowsCount() (resultRows int64) {
 	return sc.RuntimeStatsColl.GetPlanActRows(planID)
 }
 
+// UpdateDeltaForTable updates the delta info for some table.
+func (sc *StatementContext) UpdateDeltaForTable(
+	physicalTableID int64,
+	delta int64,
+	count int64,
+) {
+	sc.mu.Lock()
+	defer sc.mu.Unlock()
+	if sc.TableDeltaMap == nil {
+		sc.TableDeltaMap = make(map[int64]TableDelta)
+	}
+	item := sc.TableDeltaMap[physicalTableID]
+	item.Delta += delta
+	item.Count += count
+	item.TableID = physicalTableID
+	sc.TableDeltaMap[physicalTableID] = item
+}
+
+// ConsumeTableDelta consumes table delta info.
+func (sc *StatementContext) ConsumeTableDelta(fn func(tid int64, item TableDelta)) {
+	sc.mu.Lock()
+	defer sc.mu.Unlock()
+	if sc.TableDeltaMap == nil {
+		return
+	}
+	for id, item := range sc.TableDeltaMap {
+		fn(id, item)
+	}
+	sc.TableDeltaMap = nil
```

Comment:
nit: we can store the `sc.TableDeltaMap` to a temporary variable , then reset `sc.TableDeltaMap` to `nil`, then unlock, then iterate the temporary variable to call `fn`. So `fn` is moved out of lock scope